plugin/core/compat.gradle

```diff
@@ -0,0 +1,56 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
+String gw = Os.isFamily(Os.FAMILY_WINDOWS) ? 'gradlew.bat' : './gradlew'
+String compatClassesDestination = 'build/compat/classes'
+TaskContainer taskContainer = project.tasks
+
+File[] compatModulesDirs = rootProject.file('plugin/compat').listFiles(new FilenameFilter() {
+
+    /**
+     * We want to collect all the folders inside plugin/compat that are also standalone Gradle projects,
+     * hence we look for their settings.gradle
+     */
+    @Override
+    boolean accept(File dir, String name) {
+        return new File(dir, "$name/settings.gradle").exists()
+    }
+})
+
+Task compatClassesTask = taskContainer.create(name: 'releasePluginCompatClasses', type: DefaultTask) {
+    outputs.dir(compatClassesDestination)
+}
+
+compatModulesDirs.each { File moduleDir ->
+
+    def moduleName = moduleDir.name
+
+    def cleanModuleTask = taskContainer.create(name: "clean${moduleName.capitalize()}", type: Delete) {
+        delete new File(moduleDir, 'build')
+    }
+    taskContainer.clean.dependsOn cleanModuleTask
+
+    def compileModuleTask = taskContainer.create(name: "compile${moduleName.capitalize()}Classes", type: Exec) {
```

Comment:
we want to kick-off the compilation of the compat module, but given that is a separate Gradle project we then leverage an ad-hoc `Exec` task to run the Gradle build for that project (that will use a different version of Gradle for instance)

Reply:
same here, why do we manually need to create a `compile` task for each module?

Reply:
so, the comment above yours was my attempt at highlighting that each module is a **separate Gradle project**, that is `plugin/compat/xyz` is not in the same Gradle project as `plugin/core`.

To invoke the compile task in a separate Gradle project we are leveraging `Exec` to run a Gradle build from a Gradle build (buildception!)